packages/deploy/src/validator.ts

```diff
@@ -50,14 +53,23 @@ export function parseAndValidate(input: string): {
     if (typeof workflows === 'undefined') {
       // allow workflows to be unspecified, but ensure there is an empty
       // map to avoid errors downstream
-      doc.setIn(['workflows'], {})
-    }
-    else if (isMap(workflows)) {
+      doc.setIn(['workflows'], {});
+    } else if (isMap(workflows)) {
       for (const workflow of workflows.items) {
         if (isPair(workflow)) {
-          pushUniqueKey(workflow, (workflow as any).key.value);
-
-          validateJobs((workflow as any).value);
+          const workflowName = (workflow as any).key.value;
+          pushUniqueKey(workflow, workflowName, 'workflow');
+          const workflowValue = (workflow as any).value;
+          if (isMap(workflowValue)) {
+            jobKeys = new Set(); // Reset job keys for each workflow
```

Comment:
Instead of "resetting" the jobKeys, we should create a new array as a const. In other words, declare the array here, rather than higher up